### ISSUE TITLE
Upload AASHE 2018 Presentation information and Files

### ISSUE THREAD

**Issue by monika-aashe** [AASHE/hub#541]
We are ready for the upload! All info provided below. Goal is to send all info to have presentations uploaded in Hub around first week of November. 

[Link to the data file ](https://docs.google.com/spreadsheets/d/1BFNXqj4fB0SsMw2VZWjG66oYsogDqJHRqoD_YSh6_eY/edit#gid=2002386321)
- Make sure you are on the "IT Data" tab. 

Link to zipped folders with files - According to AI, the links should be available until middle of December: 

- [pdfs](https://files.aievolution.com/ash1801/docs/pdfs.zip)
- [ppts](https://files.aievolution.com/ash1801/docs/presentations.zip)

A few notes:

- See last few columns of "IT Data" to see which files go with which presentation. There are more files in the zipped folder than what we wish to upload (many users uploaded duplicate files, which we parsed through manually)
- All resources should be uploaded under the Permission Level "Member - AASHE Member Status Required"
- Submitter can be Monika Urbanski
- Organization names always include the AMS ID in addition to the Org name. If there is a discrepancy, please go with whatever Org ID is, since minor edits may have been submitted since the organization report was run. 
- Please upload to staging and ask Monika to test before releasing on the live server

I'm happy to answer any questions you may have!

**Comment by rerb** [comment]
Timeline is tight. I’m on vacation last week of October and first week of November so though I’m sure Tylor can do it, other priorities might clash. Maybe best to push expectations back two weeks.

**Comment by monika-aashe** [comment]
Bob and Tyler,
Everything is set for the upload into the Hub. Can you let us know if any delays are expected? From my understanding, We would like to have this task prioritized since M&M is waiting on this for some important communications. Thanks in advance!

**Comment by dodget** [comment]
I attempted to upload half of the presentations to staging today. It looks good, except for the file uploads. It appears that the upload process does not like uploading directly from url and would prefer a path from disk.

I should be able to download the zip files, and run the script referencing the directory location on my local machine.

Additional comment - While I was researching why the files appeared busted, I was looking at the two conference presentations that were already manually input. It looks like their files were busted in the same way. I downloaded the closing keynote's file to my machine from AWS, then deleted it out of the hub. I renamed the downloaded file (so there wouldn't be collision in AWS) and re-uploaded it. It seems to work fine now. I'll do the same for the opening keynote, now.

**Comment by monika-aashe** [comment]
Thanks Tylor! That is strange about the two manual uploads. I'm assuming you are talking about the two listed [here](https://hub.aashe.org/browse/types/presentation/?search=&gallery_view=list&content_type=presentation&country=&created=2018&conference_name=1#resources-panel)? If so, Andrea added those herself. The Youtube links were copied over from that site, and the presentation files are what she received from the presenters. So neither were uploaded through AI and were added through a more conventional approach. With that said, if something is "busted", is it possible the consequences might be more far-reaching, impacting other resources we've been approving? What should I be looking for? Errors when clicking on URLs or uploaded files?

**Comment by dodget** [comment]
Yup, those are the two presentations.

I can take a look at the file uploading process in the near future. Unless there is some context that I'm missing, I don't remember any issues with file uploading, so I wouldn't worry too much about far-reaching consequences.

If you are trying to find these in the wild, you would click the "File" link in "Links and Materials." If something is wrong, it will show an XML file with an AccessDenied message, and say something about "no styling has been provided." This would be a signal from AWS that either the file is bad in some way, or the permissions weren't set correctly.

If you see anything off, let me know.

**Comment by dodget** [comment]
Ready for review on staging :)

**Comment by monika-aashe** [comment]
Thanks Tylor, can you shoot me the link to the staging server?

**Comment by monika-aashe** [comment]
Thanks for the link!
https://aashe-hub-stage.herokuapp.com/

I spot checked and didn't find any issues. A few questions though:

- Does search run differently on the staging server than on the main site? It isn't working very well on the staging site, but I am not recreating the same issues on the main site right now. 
- Is there a list of the organizations that you mentioned didn't show up on the staging site so I can double-check those more closely when we add them to publication? 

I think this is pretty close to adding to the main site! No major fixes barring the search question.

**Comment by dodget** [comment]
Great! 

Yes, search doesn't work on staging. Searching requires a third-party service called 'Elastic Search,' which is not configured for staging.

Yes, I have an output of the organization IDs that were "not found" for staging. I will receive another output when the script is run in production. I'll provide the current list of ID's by email in the next few minutes.

**Comment by monika-aashe** [comment]
Good to know. 
I got your list and compared it to what is in the AMS. It does appear that the organizations on the list that don't show up on staging are ones that were added to the AMS since May 2018. 

Do you still anticipate this not being an issue on the production site? If that's the case, then I think it's safe to load to production and I'll do some spot checks once there.

**Comment by dodget** [comment]
Great. I don't anticipate any issues in production. As a test, I just went to the submit resource in production, and when adding an author, all of the orgs in question appear just fine. I did the same on the staging server, and the orgs in question do not appear, which is exactly what we'd expect.

I'll be able to run the script sometime after lunch. I'll alert you when it's finished, and provide any relevant outputs.

**Comment by dodget** [comment]
Conference presentations have been successfully uploaded.

There was only one line output for "organization ID not found." That is for ID 8284, which was supposed to correspond to "[CU Boulder Tests Behavior-Tracking Platforms..](https://hub.aashe.org/browse/types/presentation/?search=&gallery_view=list&content_type=presentation&tagfilter=behavior-change&organizations=18903&country=&published=2018&order=-published)." It looks like this ID used to correspond to PIPS Rewards, which is no longer appearing in Membersuite.

I've been checking if the resources are returned in searches and filtering, and its all looking good!

Many thanks to you, and anyone who worked on it, for such an excellent spreadsheet. It makes this process very smooth.

**Comment by monika-aashe** [comment]
Tylor,
We very recently merged PIPs rewards with its parent company, 3P Partners, Inc. It looks like I remembered to update the Author Org but not the main resource org field. Sorry! I just fixed it manually. 

Thanks for the kudos! It's a time intensive process but hopefully worth it for our members. I will do a few spot checks before sending the rest of the team the good news that this is ready to announce. 

Thanks for your quick and quality work on this!
